packages/twenty-front/src/modules/apollo/optimistic-effect/utils/triggerUpdateRecordOptimisticEffect.ts

```diff
@@ -117,9 +117,18 @@ export const triggerUpdateRecordOptimisticEffect = ({
           });
         }
 
+        const totalCount = readField<number | undefined>(
+          'totalCount',
+          rootQueryConnection,
+        );
+
         return {
           ...rootQueryConnection,
           edges: rootQueryNextEdges,
+          totalCount: isDefined(totalCount)
+            ? totalCount -
+              (rootQueryCurrentEdges.length - rootQueryNextEdges.length)
+            : undefined,
```

Comment:
It was intentional because 
1. for removal, `rootQueryCurrentEdges.length` > `rootQueryNextEdges.length` thus `totalCount` decreases
2. for addition, `rootQueryCurrentEdges.length` < `rootQueryNextEdges.length` thus `totalCount` increases